Libraries/DirectMLX.h

```diff
@@ -722,10 +745,14 @@ namespace dml
             // number of input nodes on the graph (e.g. in the case of unused empty inputs), but never smaller.
             assert(inputCount == 0 || inputCount >= graph.inputCount);
 
-            std::vector<DML_GRAPH_NODE_DESC> graphNodes(graph.nodes.size());
-            for (size_t i = 0; i < graphNodes.size(); ++i)
+            std::vector<DML_GRAPH_NODE_DESC> graphNodes(graph.operatorNodes.size() + graph.constantNodes.size());
+            for (size_t i = 0; i < graph.operatorNodes.size(); ++i)
+            {
+                graphNodes[i] = { DML_GRAPH_NODE_TYPE_OPERATOR, &graph.operatorNodes[i] };
+            }
+            for (size_t i = 0; i < graph.constantNodes.size(); ++i)
             {
-                graphNodes[i] = { DML_GRAPH_NODE_TYPE_OPERATOR, &graph.nodes[i] };
+                graphNodes[i + graph.operatorNodes.size()] = {DML_GRAPH_NODE_TYPE_CONSTANT, &graph.constantNodes[i]};
```

Comment:
Another approach which is a little more immune to reordering later and less indexing to reason about inside the loops is to slice the arrays early. e.g.

```c++
            std::vector<DML_GRAPH_NODE_DESC> graphNodes(graph.operatorNodes.size() + graph.constantNodes.size());
            auto operatorNodes = std::span<DML_GRAPH_NODE_DESC>(graphNodes).first(graph.operatorNodes.size());
            auto constantNodes = std::span<DML_GRAPH_NODE_DESC>(graphNodes).last(graph.constantNodes.size());

            for (size_t i = 0; i < graph.operatorNodes.size(); ++i)
            {
                operatorNodes[i] = {DML_GRAPH_NODE_TYPE_OPERATOR, &graph.operatorNodes[i]};
            }
            for (size_t i = 0; i < graph.constantNodes.size(); ++i)
            {
                constantNodes[i] = {DML_GRAPH_NODE_TYPE_CONSTANT, &graph.constantNodes[i]};
            }
```
Downside is more code though :(. Shrug, take it or leave it.